testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/servlet/preservepath/PreservePathTestCase.java

```diff
@@ -95,33 +107,66 @@ public static WebArchive deployment() {
 
    @Test
    public void testPreservePath() throws Exception {
-      setPreservePathOnForward("true");
-
-      HttpClient httpclient = HttpClientBuilder.create().build();
-      HttpGet httpget = new HttpGet(url.toString() + "/test?path="+ URLEncoder.encode(tempDir));
-      HttpResponse response = httpclient.execute(httpget);
-
-      Thread.sleep(100);
-      FileReader fr = new FileReader(new File(tempDir + "/output.txt"));
-      Assert.assertEquals("servletPath: /test", new BufferedReader(fr).readLine());
+      runTestPreservePath(true);
    }
 
    @Test
    public void testDontPreservePath() throws Exception {
-      setPreservePathOnForward("false");
+      runTestPreservePath(false);
+   }
+
+   @Test
+   public void testDefaultPreservePath() throws Exception {
+      runTestPreservePath(null);
+   }
+
+   private void runTestPreservePath(Boolean preservePathOnForward) throws Exception {
+      setPreservePathOnForward(preservePathOnForward);
 
       HttpClient httpclient = HttpClientBuilder.create().build();
       HttpGet httpget = new HttpGet(url.toString() + "/test?path="+ URLEncoder.encode(tempDir));
       HttpResponse response = httpclient.execute(httpget);
 
-      Thread.sleep(100);
-      FileReader fr = new FileReader(new File(tempDir + "/output.txt"));
-      Assert.assertEquals("servletPath: /preserve-path.jsp", new BufferedReader(fr).readLine());
+      long end = System.currentTimeMillis() + TIMEOUT;
+      File file = new File(tempDir + "/output.txt");
+      while (!file.exists() && System.currentTimeMillis() < end) {
+         Thread.sleep(100);
+      }
+      Assert.assertTrue(file + " was not created within the timeout", file.exists());
```

Comment:
Based on execution on my own machine, there seems to be not enough time for server to complete write into 'output.txt' file before its content is checked - when I enforce longer sleep here (5sec), the tests pass successfully for me.

Reply:
@spyrkob, I've created following PR against this branch, which looks like helps this test to be more stable https://github.com/spyrkob/wildfly/pull/1. Feel free to incorporate into your branch.

Reply:
Thanks @jstourac, merged it in